jax/experimental/sparse/ad.py

```diff
@@ -66,9 +66,16 @@ def reconstruct(i, grad_out):
       f_recons = jax.vmap(f_recons)
     return f_recons(grad_out)
 
+  def tree_is_none(tree):
+    return tree_util.tree_all(tree_util.tree_map(lambda x: x is None, tree))
+
   def postprocess_gradients(grads_out):
-    out = [reconstruct(*args) for args in safe_zip(argnums_flat1, grads_out)]
-    return out[0] if isinstance(argnums, int) else out
+    grads = [reconstruct(*args) for args in safe_zip(argnums_flat1, grads_out)]
+    leaf_grads = [grads.pop(0) if l_idx in argnums_flat1 else None for l_idx in
```

Comment:
This is confusing (and potentially error-prone, if `grads` is not the expected size). What about something like this:
```python
leaf_grads = [None] * tree1.num_leaves
for i, grad in safe_zip(argnums_flat_1, grads):
  leaf_grads[i] = grad
```

Reply:
This is a good improvement; I've incorporated it.